hpssa/hpssa.py

```diff
@@ -704,6 +704,23 @@ def get_pd_by_index(self, slot, idx):
         pd_info = adapter['drives'][idx]
         return self.assemble_id(pd_info)
 
+    def _set_hba_mode(self, slot, state):
+        LOG.info('Setting %s mode on controller in Slot %s',
+                 'HBA' if state == 'on' else 'RAID', slot)
+        cmd = 'ctrl slot={} modify hbamode={} forced'.format(slot, state)
+        result = self.run(cmd)
+        LOG.warning('Command sent. The device must be rebooted for this change '
+                    'to take effect')
+        return result
+
+    def enable_hba_mode(self, slot):
+        """ Sets the controller in HBA mode """
+        return self._set_hba_mode(slot, 'on')
```

Comment:
Just out of curiosity, why are you using strings for on and off instead of booleans? (It doesn't matter much since it's a private method, obviously)

Reply:
The controller takes on/off on the command line. I could have used True / False and `'on' if True else 'off'` .. but decided to just pass it through.